lib/uri.js

```diff
@@ -27,6 +27,12 @@ function URI(uri, params, asPattern) {
         }
         this.path = utils.parsePath(uri, asPattern);
     } else if (Array.isArray(uri)) {
+        if (!asPattern) {
```

Comment:
asPattern currently supports passing in pre-parsed objects, and we don't want to stringify those.